internal/venice-common/src/test/java/com/linkedin/venice/meta/TestHelixReadWriteSchemaRepository.java

```diff
@@ -0,0 +1,58 @@
+package com.linkedin.venice.meta;
+
+import com.linkedin.venice.helix.HelixAdapterSerializer;
+import com.linkedin.venice.helix.HelixReadWriteSchemaRepository;
+import com.linkedin.venice.helix.HelixSchemaAccessor;
+import com.linkedin.venice.schema.writecompute.DerivedSchemaEntry;
+import com.linkedin.venice.system.store.MetaStoreWriter;
+import com.linkedin.venice.utils.Pair;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.avro.Schema;
+import org.apache.helix.zookeeper.impl.client.ZkClient;
+import org.mockito.Mockito;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+
+public class TestHelixReadWriteSchemaRepository {
+  private HelixReadWriteSchemaRepository helixReadWriteSchemaRepository;
+  private final HelixSchemaAccessor accessor = Mockito.mock(HelixSchemaAccessor.class);
+  private final ReadWriteStoreRepository storeRepository = Mockito.mock(ReadWriteStoreRepository.class);
+  private final ZkClient zkClient = Mockito.mock(ZkClient.class);
+  private final HelixAdapterSerializer adapter = Mockito.mock(HelixAdapterSerializer.class);
+  private final MetaStoreWriter metaStoreWriter = Mockito.mock(MetaStoreWriter.class);
+
+  @BeforeMethod
+  void setUp() {
+    String clusterName = "testCluster";
+    this.helixReadWriteSchemaRepository = new HelixReadWriteSchemaRepository(
+        storeRepository,
+        zkClient,
+        adapter,
+        clusterName,
+        Optional.of(metaStoreWriter));
+    this.helixReadWriteSchemaRepository.setAccessor(accessor);
```

Comment:
Just a nitpick but... can we overload the constructor to inject our mock, rather than adding a setter? This way the schema accessor can remain final within the schema repo.

Reply:
Sure, updated.